Issue #2417: ITPParser should not add angles to SETTLEd waters.

BFedder opened the pull request:
Fixes #2417

Changes made in this Pull Request:
 ITPParser no longer adds angles for water molecules with SETTLEd
   geometry. The tests were modified to reflect this.

Changes:
**MDAnalysis/topology/ITPParser.py**
- Molecule.parse_settles changed to no longer add angle

**MDAnalysisTests/topology/test_itp.py**
- TestITPNoKeywords now expects 0 angles
- TestITPKeywords changed to continue to expect 1 angle
- def test_angles_values moved from TestITPNoKeywords to TestITPKeywords
- TestNestedIfs now expects 0 angles
- TestReadTop's topology file has 3 SETTLEd water molecules and now correctly expects three fewer angles than before


PR Checklist
------------
 - [x] Tests?
 - [ ] Docs?
 - [x] CHANGELOG updated?
 - [x] Issue raised/referenced?

BFedder committed:
Issue #2417: ITPParser, no angle for SETTLEd water

ITPParser no longer adds angles for water molecules with SETTLEd
geometry. The tests were modified to reflect this.
Changes:
  MDAnalysis/topology/ITPParser.py
    *  Molecule.parse_settles changed to no longer add angle

  MDAnalysisTests/topology/test_itp.py
    *  TestITPNoKeywords now expects 0 angles
    *  TestITPKeywords changed to continue to expect 1 angle
    *  def test_angles_values moved from TestITPNoKeywords to
         TestITPKeywords
    *  TestNestedIfs now expects 0 angles
    *  TestReadTop's topology file has 3 SETTLEd water molecules
         and now correctly expects three fewer angles than before

BFedder committed:
added changes to AUTHORS and CHANGELOG

BFedder committed:
fixed typo

pep8speaks commented:
Hello @BFedder! Thanks for updating this PR. We checked the lines you've touched for [PEP 8](https://www.python.org/dev/peps/pep-0008) issues, and found:





There are currently no PEP 8 issues detected in this Pull Request. Cheers! :beers: 

##### Comment last updated at 2022-03-17 10:36:04 UTC

orbeckst reviewed (commented):
This is already looking very good. Can you please look at the PEP8 comments?

I'd like to have @jbarnoud and/or @lilyminium have a look, too.

If you haven't done so already, please introduce yourself on the [developer mailing list](https://groups.google.com/g/mdnalysis-devel) so that we have a current email address from you. Thank you!

jbarnoud commented on package/CHANGELOG:
```diff
@@ -22,6 +22,9 @@ Fixes
 Changes
+  * ITPParser no longer adds an angle for water molecules that have the SETTLE
+    constraint applied to them. This mirrors the way the constraint is handled
+    in AMBER. (Issue #2417, PR #)
```
```suggestion
    in AMBER. (Issue #2417, PR #3564)
```

jbarnoud reviewed (approved):
Just a tiny fix for the changelog. Otherwise, looks good to me. This looks like a very clean fix.

BFedder committed:
Update ITPParser.py

jbarnoud reviewed (changes requested):
After a second look, it would be good to add a `.. versionchanged:` notice in the docstring of the `ITPParser` class. Just that small documentation fix and it is good to go.

BFedder committed:
Update test_itp.py

BFedder committed:
Update package/CHANGELOG

Co-authored-by: Jonathan Barnoud <jonathan@barnoud.net>

BFedder committed:
Update ITPParser.py

BFedder commented:
Hello and thanks! 
@orbeckst, I have now addressed the PEP8 issues and introduced myself on the mailing list. 
@jbarnoud, thanks for the updated changelog - I have added the PR number now, and also added a `.. versionchanged::` note to `ITPParser`.

jbarnoud reviewed (approved):
Thank you! And congratulation on your first merged contribution :smile:
